### PR TITLE
Add task to current sprint

### PR DESCRIPTION
**Pull request by duanxianbo**

### CODE REVIEW

File: apps/github/app.go [hunk 1]
```diff
@@ -17,13 +17,17 @@ import (
 	"strings"
 	"time"
 
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
```
Comment (naturetime10) [inline apps/github/app.go]:
```suggestion
```

File: apps/github/app.go [hunk 2]
```diff
@@ -17,13 +17,17 @@ import (
...
 	cloudAPI "github.com/teamyapp/cloud/app/api"
 	cloudProto "github.com/teamyapp/cloud/app/api/proto"
 	"github.com/teamyapp/cloud/libs/runner"
 	"github.com/teamyapp/teamy-backend/apps/dao"
 	"github.com/teamyapp/teamy-backend/apps/entity"
 	"github.com/teamyapp/teamy-backend/core/api"
 	"github.com/teamyapp/teamy-backend/core/api/proto"
+
```
Comment (naturetime10) [inline apps/github/app.go]:
```suggestion
```

Comment (naturetime10) [reply apps/github/app.go]: Please run `goimports -w apps` and `goimports -w core` to format the code

File: apps/github/app.go [hunk 3]
```diff
@@ -197,7 +201,8 @@ func (a App) onEventNotify(w http.ResponseWriter, r *http.Request) {
 	deliveryID := r.Header.Get("X-GitHub-Delivery")
 	evtType := r.Header.Get("X-GitHub-Event")
 	log.Printf("Github event received: deliveryID=%s, event=%s\n", deliveryID, evtType)
-	err = a.processEvent(r.Context(), eventType(evtType), buf)
+	ct, _ := context.WithTimeout(context.Background(), time.Minute*60) // TODO
```
Comment (naturetime10) [inline apps/github/app.go]: Move `time.Minute*60` to the top of the file as a constant or create environmental variable to configure it. In production, a request cannot take 1 hour to finish.

File: apps/github/app.go [hunk 4]
```diff
@@ -499,13 +504,26 @@ func (a App) tryCreateTaskForPullRequestReviewer(
...
 	codeReview = entity.GithubCodeReview{
 		GithubPullRequestNodeID:  githubPullRequestNodeID,
 		InternalCodeReviewTaskID: createdTaskID,
 		GithubReviewerID:         githubReviewerID,
 		Round:                    1,
 	}
-	return a.githubCodeReviewDao.CreateCodeReview(codeReview)
+
+	err = a.githubCodeReviewDao.CreateCodeReview(codeReview)
+
```
Comment (naturetime10) [inline apps/github/app.go]:
```suggestion
```

File: apps/github/app.go [hunk 5]
```diff
@@ -544,6 +562,34 @@ func (a App) createCodeReviewTask(ct context.Context, teamID uint64, pullRequest
 	return createTaskRes.TaskId, nil
 }
 
+func (a App) tryAddTaskToCurrentSprint(ct context.Context, teamID uint64, taskID uint64) error {
+	getCurrentSprintReq := &proto.GetCurrentSprintRequest{TeamId: teamID}
+
```
Comment (naturetime10) [inline apps/github/app.go]:
```suggestion
```

File: apps/github/app.go [hunk 6]
```diff
@@ -544,6 +562,34 @@ func (a App) createCodeReviewTask(ct context.Context, teamID uint64, pullRequest
...
+	getCurrentSprintRes, err := a.teamyClientRegistry.SprintClient().GetCurrentSprint(ct, getCurrentSprintReq)
+
```
Comment (naturetime10) [inline apps/github/app.go]:
```suggestion
```

File: apps/github/app.go [hunk 7]
```diff
@@ -544,6 +562,34 @@ func (a App) createCodeReviewTask(ct context.Context, teamID uint64, pullRequest
...
+	if err != nil {
+		st := status.Convert(err)
+
```
Comment (naturetime10) [inline apps/github/app.go]:
```suggestion
```

File: apps/github/app.go [hunk 8]
```diff
@@ -544,6 +562,34 @@ func (a App) createCodeReviewTask(ct context.Context, teamID uint64, pullRequest
...
+		if st.Code() == codes.NotFound {
+			return nil
+		}
+
+		log.Println(err)
+		return err
+	}
+
+	addTaskToSprintReq := &proto.AddTaskToSprintRequest{TaskId: taskID, SprintId: getCurrentSprintRes.Id}
+
```
Comment (naturetime10) [inline apps/github/app.go]:
```suggestion
```

File: apps/github/app.go [hunk 9]
```diff
@@ -544,6 +562,34 @@ func (a App) createCodeReviewTask(ct context.Context, teamID uint64, pullRequest
...
+	_, err = a.teamyClientRegistry.SprintClient().AddTaskToSprint(ct, addTaskToSprintReq)
+
```
Comment (naturetime10) [inline apps/github/app.go]:
```suggestion
```

File: apps/github/app.go [hunk 10]
```diff
@@ -544,6 +562,34 @@ func (a App) createCodeReviewTask(ct context.Context, teamID uint64, pullRequest
...
+	if err != nil {
+		log.Println(err)
+		return err
+	}
+
+	return nil
```
Comment (naturetime10) [inline apps/github/app.go]:
```suggestion
	return err
```

File: core/api/sprint_rpc.go [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+
+import (
+	"context"
+	_ "embed"
+	"errors"
+	"log"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
```
Comment (naturetime10) [inline core/api/sprint_rpc.go]:
```suggestion
```

File: core/api/sprint_rpc.go [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+	"github.com/teamyapp/cloud/libs/runner"
+	"github.com/teamyapp/teamy-backend/core/api/proto"
+	"github.com/teamyapp/teamy-backend/core/service"
+
```
Comment (naturetime10) [inline core/api/sprint_rpc.go]:
```suggestion
```

Comment (naturetime10) [reply core/api/sprint_rpc.go]: Please format code with `goimports -w core`

File: core/service/sprint.go [hunk 13]
```diff
@@ -50,6 +50,40 @@ func (s Sprint) FindTasksInSprint(
 	return tasks, nil
 }
 
+func (s Sprint) FindCurrentSprint(ct context.Context, teamID uint64) (entity.Sprint, error) {
+	sprints, err := s.sprintDao.FindAllSprints()
```
Comment (naturetime10) [inline core/service/sprint.go]:
```suggestion
	sprints, err := s.sprintDao.FindSprintsByTeamID(teamID)
```

File: core/service/sprint.go [hunk 14]
```diff
@@ -50,6 +50,40 @@ func (s Sprint) FindTasksInSprint(
...
+		return entity.Sprint{}, err
+	}
+
+	now := time.Now()
+
+	sprints = collect.Filter(sprints, func(sprint entity.Sprint) bool {
+		if teamID != sprint.OwningTeamID {
+			return false
+		}
+
```
Comment (naturetime10) [inline core/service/sprint.go]:
```suggestion
```

File: core/service/sprint.go (hunk 13 above)
Comment (naturetime10) [reply core/service/sprint.go]: This DAO API is more efficient as we are fetching less data from DB

File: core/service/sprint.go [hunk 15]
```diff
@@ -50,6 +50,40 @@ func (s Sprint) FindTasksInSprint(
...
+		if now.Before(sprint.StartAt) || now.After(sprint.EndAt) {
+			return false
+		}
+
+		return true
+	})
+
```
Comment (naturetime10) [inline core/service/sprint.go]:
```suggestion
```

File: core/service/sprint.go [hunk 16]
```diff
@@ -50,6 +50,40 @@ func (s Sprint) FindTasksInSprint(
...
+	if len(sprints) < 1 {
+		err := ErrNotFound("no active sprint found")
```
Comment (naturetime10) [inline core/service/sprint.go]:
```suggestion
		err := ErrNotFound("no current sprint found: teamID=%v, currentTime=%v", now.UTC())
```

**Review by naturetime10 — CHANGES_REQUESTED** [review]

File: apps/github/app.go (hunk 3 above)
Comment (duanxianbo) [reply apps/github/app.go]: done

**Review by naturetime10 — APPROVED** [review]

**Pull request merged**